### PR TITLE
Edit Owner

### LINKED ISSUE #524: Edit Owner : User picker
To edit the owner of a content, we should display a control that allows to select a user.

If the content definition setting 'allow to edit the owner' is enabled, the owner name would be displayed as a label and a change link would open a popup to select a user in a filterable list.

### PR DESCRIPTION
**Pull request by agriffard**
Check if the owner is an existing user.

Fixes #524

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Contents/Drivers/OwnerEditorDriver.cs [hunk 1]
```diff
@@ -70,7 +78,16 @@ public class OwnerEditorDriver : ContentPartDisplayDriver<CommonPart>
 
                 if (!string.IsNullOrEmpty(part.ContentItem.Owner) || !context.IsNew)
                 {
-                    part.ContentItem.Owner = model.Owner;
+                    var newOwner = await _userManager.FindByNameAsync(model.Owner);
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Contents/Drivers/OwnerEditorDriver.cs]: Add security checks

https://github.com/OrchardCMS/Orchard/blob/dev/src/Orchard.Web/Core/Common/OwnerEditor/OwnerEditorDriver.cs#L37

Comment (agriffard) [reply src/OrchardCore.Modules/OrchardCore.Contents/Drivers/OwnerEditorDriver.cs]: There is already that kind of check:

            if (currentUser == null || !(await _authorizationService.AuthorizeAsync(currentUser, StandardPermissions.SiteOwner, part)))
            {
                return null;
            }

**Comment by sebastienros** [comment]
I am glad we are consistent between what we do and what we say we should do then.

**Comment by PinpointTownes** [comment]
Having to reference Identity in a low-level block like Content seems meh to me.

**Comment by sebastienros** [comment]
> like Content

But content has the notion of permissions and users all over. What's the alternative?

**Comment by PinpointTownes** [comment]
Use the `IUserService` abstraction?

**Comment by sebastienros** [comment]
https://github.com/OrchardCMS/OrchardCore/pull/1896

**Pull request merged**